src/tools/crawling_tools.py

```
@@ -0,0 +1,703 @@
+            "total_chunks": results["total_chunks"],
+            "failed_urls": results["failed_urls"],
+        }
+
+    except requests.RequestException as e:
+        raise NetworkError(f"Failed to fetch text file: {e}") from e
+    except Exception as e:
+        raise ProcessingError(f"Error processing text file: {e}") from e
+
+
+async def _crawl_sitemap(
+    crawler: AsyncWebCrawler, url: str, supabase_client, max_depth: int, max_concurrent: int
```

Pylint (3.3.7)

[warning] Unused argument 'max_depth'

(W0613)